src/category_theory/limits/shapes/images.lean

```diff
@@ -786,12 +786,20 @@ instance has_strong_epi_images_of_has_pullbacks_of_has_equalizers [has_pullbacks
   [has_equalizers C] : has_strong_epi_images C :=
 { strong_factor_thru_image := λ X Y f,
   { epi := by apply_instance,
-    has_lift := λ A B x y h h_mono w, arrow.has_lift.mk
-    { lift := image.lift
-      { I := pullback h y,
-        m := pullback.snd ≫ image.ι f,
-        m_mono := by exactI mono_comp _ _,
-        e := pullback.lift _ _ w } ≫ pullback.fst } } }
+    llp := begin
+      introsI A B h h_mono,
+      constructor,
+      intros x y sq,
+      exact comm_sq.has_lift.mk'
+      { l := image.lift
+          { I := pullback h y,
+            m := pullback.snd ≫ image.ι f,
+            m_mono := by exactI mono_comp _ _,
+            e := pullback.lift _ _ sq.w } ≫ pullback.fst,
+        fac_left' := by simp only [image.fac_lift_assoc, pullback.lift_fst],
+        fac_right' := by { ext, simp only [sq.w, category.assoc,
+          image.fac_lift_assoc, pullback.lift_fst_assoc], }, },
+    end, }, }
```

Comment:
This got about twice as long. Should there be a custom constructor maybe?

Reply:
Thanks for the review. I have added custom constructor, which slightly reduced the length of a few proofs.